File: ios/App/App/KeepAwake.m
#import <Foundation/Foundation.h>
#import <Capacitor/Capacitor.h>

CAP_PLUGIN(KeepAwake, "KeepAwake",
           CAP_PLUGIN_METHOD(keepAwake, CAPPluginReturnPromise);
           CAP_PLUGIN_METHOD(allowSleep, CAPPluginReturnPromise);
           )

File: ios/App/App/KeepAwake.swift
import Foundation
import Capacitor

@objc(KeepAwake)
public class KeepAwake: CAPPlugin {

    @objc func keepAwake(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            if(!UIApplication.shared.isIdleTimerDisabled) {
                UIApplication.shared.isIdleTimerDisabled = true
            }
            call.resolve()
        }
    }

    @objc func allowSleep(_ call: CAPPluginCall) {
        DispatchQueue.main.async {
            if(UIApplication.shared.isIdleTimerDisabled) {
                UIApplication.shared.isIdleTimerDisabled = false
            }
            call.resolve()
        }
    }
}
